generator.rb

```diff
@@ -0,0 +1,73 @@
+# Initiate array of vowels accessible everywhere
+VOWELS = ["a", "e", "i", "o", "u"]
+
+# Check if character is a vowel
+def is_vowel? (letter)
+    if VOWELS.include? (letter)
+        true
+    else
+        false
+    end
+end
+
+# Get user input to form new word
+def run_generator
+
+  puts "What is the first word to combine?"
+  word_A = gets.chomp
+  until word_A.size >= 2 
+    puts "Your word should have at least 2 letters."
+    word_A = gets.chomp
+  end
+
+  # First segment of new word
+  new_A = ""
+    
+    # If word has no vowels, use entire word
+  if word_A.count("aeiou") == 0
+    new_A = word_A
+  else
+    # Select segment from first character to last vowel, exluding last vowel
+    i = word_A.length - 1
+    word_A.reverse.each_char do |v|
+      if is_vowel?(v) == true 
+        new_A = word_A[0...i]
+        break
+      end
+      i -= 1
+    end
+  end
+
+  # Get input for second part of new word
+  puts "What is the second word to combine?"
+  word_B = gets.chomp
+  until word_B.size >= 2
+    puts "Your word should have at least 2 letters."
+    word_B = gets.chomp
```

Comment:
Your code to get the second word is almost exactly the same as the code to get the first word. Could you DRY this up using another method?